### ISSUE TITLE
dynamodb:GetItem fails with socket.gaierror using AsyncioBotocore

### ISSUE THREAD

**Issue by adamrothman** [quantmind/pulsar-cloud#17]
```python
Traceback (most recent call last):
  File "./ddbench.py", line 564, in <module>
    loop.run_until_complete(main(args.n))
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/asyncio/base_events.py", line 387, in run_until_complete
    return future.result()
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/asyncio/tasks.py", line 241, in _step
    result = coro.throw(exc)
  File "./ddbench.py", line 554, in main
    pulsar_avg = await time_pulsar(n, kwargses)
  File "./ddbench.py", line 531, in time_pulsar
    in kwargses
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/asyncio/futures.py", line 361, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/asyncio/tasks.py", line 296, in _wakeup
    future.result()
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/asyncio/tasks.py", line 239, in _step
    result = coro.send(None)
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/site-packages/cloud/asyncbotocore/client.py", line 73, in _make_api_call
    operation_model, request_dict)
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/site-packages/cloud/asyncbotocore/endpoint.py", line 86, in _send_request
    success_response, exception):
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/site-packages/cloud/asyncbotocore/endpoint.py", line 155, in _needs_retry
    caught_exception=caught_exception, request_dict=request_dict)
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/site-packages/botocore/hooks.py", line 227, in emit
    return self._emit(event_name, kwargs)
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/site-packages/botocore/hooks.py", line 210, in _emit
    response = handler(**kwargs)
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/site-packages/botocore/retryhandler.py", line 183, in __call__
    if self._checker(attempts, response, caught_exception):
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/site-packages/botocore/retryhandler.py", line 251, in __call__
    caught_exception)
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/site-packages/botocore/retryhandler.py", line 269, in _should_retry
    return self._checker(attempt_number, response, caught_exception)
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/site-packages/botocore/retryhandler.py", line 317, in __call__
    caught_exception)
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/site-packages/botocore/retryhandler.py", line 223, in __call__
    attempt_number, caught_exception)
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/site-packages/botocore/retryhandler.py", line 359, in _check_caught_exception
    raise caught_exception
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/site-packages/cloud/asyncbotocore/endpoint.py", line 114, in _get_response
    headers=headers, stream=True, verify=self.verify)
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/site-packages/pulsar/apps/http/__init__.py", line 988, in _request
    conn = await pool.connect()
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/site-packages/pulsar/async/clients.py", line 95, in connect
    connection = await self._get()
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/site-packages/pulsar/async/clients.py", line 137, in _get
    connection = await self._get()
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/site-packages/pulsar/async/clients.py", line 131, in _get
    connection = await self._creator()
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/site-packages/pulsar/async/clients.py", line 269, in create_connection
    protocol_factory, host, port, **kw)
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/asyncio/base_events.py", line 651, in create_connection
    infos = f1.result()
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/Users/adam/.pyenv/versions/3.5.2/lib/python3.5/socket.py", line 732, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno 8] nodename nor servname provided, or not known
```

And for reference, the code I'm trying to run:

```python
async def time_pulsar(n, kwargses):
    client = AsyncioBotocore('dynamodb')

    total = 0
    for _ in range(n):
        start = time.perf_counter()
        await asyncio.gather(*[
            client.get_item(**kwargs)
            for kwargs
            in kwargses
        ])
        delta = time.perf_counter() - start
        total += delta

    client.close()
    return total / n
```

**Comment by lsbardel** [comment]
how large is `n` and len of `kwargses`, approximately?

**Comment by adamrothman** [comment]
That particular trace was with n = 5 and 100 simultaneous GetItem calls.

**Comment by lsbardel** [comment]
This error was caused by the way pulsar Http client treated streaming requests (all botocore requests are executed in streaming mode).

However, `dynamodb` always returns body data rather than streaming output.
In other words there is no need to treat the response as streaming response and therefore remove the connection from the connection pool.

I've fixed this use case in pulsar and released pulsar 1.5.5. Try again with that version.
